web/modules/custom/joinup_invite/src/Form/InviteToGroupForm.php

```diff
@@ -51,46 +59,71 @@ public static function create(ContainerInterface $container) {
    * {@inheritdoc}
    */
   public function getFormId() {
-    return 'invite_facilitators_form';
+    return 'invite_to_group_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getSubmitButtonText(): TranslatableMarkup {
+    return $this->t('Add members');
   }
 
   /**
    * {@inheritdoc}
    */
-  protected function getSubmitButtonText() : TranslatableMarkup {
-    return $this->t('Add facilitators');
+  protected function getCancelButtonUrl(): Url {
+    return new Url('entity.rdf_entity.member_overview', [
+      'rdf_entity' => $this->rdfEntity->id(),
+    ]);
   }
 
   /**
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state, RdfInterface $rdf_entity = NULL) {
-    $form_state->set('group', $rdf_entity);
+    $this->rdfEntity = $rdf_entity;
+
+    $form['role'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Role'),
+      '#required' => TRUE,
+      '#options' => [
+        'member' => $this->t('Member'),
+        'facilitator' => $this->t('Facilitator'),
+      ],
+      '#default_value' => 'member',
+    ];
+
     return parent::build($form, $form_state);
   }
 
   /**
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $users = array_filter($form_state->getValue('users'));
-    $group = $form_state->get('group');
-    $role_id = $group->getEntityTypeId() . '-' . $group->bundle() . '-facilitator';
-    $facilitator_role = $this->entityTypeManager->getStorage('og_role')->load($role_id);
-
-    foreach ($users as $uid) {
-      $user = $this->entityTypeManager->getStorage('user')->load($uid);
-      $membership = $this->ogMembershipManager->getMembership($group, $user);
+    $users = $this->getUserList($form_state);
+    $role_id = implode('-', [
```

Comment:
Is it generally the best option to use an implode to concatenate 3 pieces of known strings in a static array?
Just asking, why not `$role_id = $this->rdf_entity->getEntityTypeId() . '-' . $this->rdf_entity->bundle() . '-' . $form_state->getValue('role');` ?

Reply:
I find it more clear to read.